ghcide/session-loader/Development/IDE/Session.hs

```diff
@@ -810,56 +774,132 @@ setNameCache :: IORef NameCache -> HscEnv -> HscEnv
 #endif
 setNameCache nc hsc = hsc { hsc_NC = nc }
 
+pprHomeUnitGraph :: HomeUnitGraph -> Compat.SDoc
+pprHomeUnitGraph unitEnv = Compat.vcat (map (\(k, v) -> pprHomeUnitEnv k v) $ Map.assocs $ unitEnv_graph unitEnv)
+
+pprHomeUnitEnv :: UnitId -> HomeUnitEnv -> Compat.SDoc
+pprHomeUnitEnv uid env =
+  Compat.ppr uid Compat.<+> Compat.text "(flags:" Compat.<+> Compat.ppr (homeUnitId_ $ homeUnitEnv_dflags env) Compat.<+> Compat.text "," Compat.<+> Compat.ppr (fmap homeUnitId $ homeUnitEnv_home_unit env) Compat.<+> Compat.text ")" Compat.<+> Compat.text "->"
+  Compat.$$ Compat.nest 4 (pprHPT $ homeUnitEnv_hpt env)
+
+
 -- | Create a mapping from FilePaths to HscEnvEqs
 newComponentCache
```

Comment:
Outdated Docs, right? Either way, it would be nice if you can expand a little bit on the purpose of this function.